pkg/util/allocationfilterutil/v2/parser_test.go

```diff
@@ -0,0 +1,298 @@
+package allocationfilterutil
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/kubecost/opencost/pkg/kubecost"
+)
+
+func allocGenerator(props kubecost.AllocationProperties) kubecost.Allocation {
+	a := kubecost.Allocation{
+		Properties: &props,
+	}
+
+	a.Name = a.Properties.String()
+	return a
+}
+
+func TestParse(t *testing.T) {
+	// TODO: unallocated cases
+	cases := []struct {
```

Comment:
Once we make a decision about `services!:"__unallocated__"` I will add more test cases.